test/test_hookcmds.py

```diff
@@ -0,0 +1,803 @@
+def test_relation_model_get(run: Run, id: int | None):
+    data = {'uuid': str(uuid.uuid4())}
+    cmd = ['relation-model-get', '--format=json']
+    if id is not None:
+        cmd.extend(['-r', str(id)])
+    run.handle(cmd, stdout=json.dumps(data))
+    result = hookcmds.relation_model_get(id=id)
+    assert str(result.uuid) == data['uuid']
+
+
+@pytest.mark.parametrize('id', [None, 123])
+@pytest.mark.parametrize('app', [False, True])
+def test_relation_set(run: Run, mock_file: NamedTemporaryFile, id: int | None, app: bool):
+    cmd = ['relation-set']
+    if app:
+        cmd.append('--app')
+    if id is not None:
+        cmd.extend(['-r', str(id)])
+    cmd.extend(['--file', '-'])
+    run.handle(cmd)
+    hookcmds.relation_set({'foo': 'bar'}, id=id, app=app)
+    assert run.calls[0].stdin == 'foo: bar\n'
+
+
+def test_resource_get(run: Run):
+    run.handle(['resource-get', 'res'], stdout='/path/to/resource')
+    result = hookcmds.resource_get('res')
+    assert str(result) == '/path/to/resource'
+
+
+def test_secret_add(run: Run, mock_temp_dir: str):
+    run.handle(['secret-add', f'foo#file={mock_temp_dir}/foo'], stdout='secretid')
+    result = hookcmds.secret_add({'foo': 'bar'})
+    assert result == 'secretid'
+
+
+@pytest.mark.parametrize('owner', ['application', 'unit'])
+def test_secret_add_with_metadata(
+    run: Run, mock_temp_dir: str, owner: Literal['application', 'unit']
+):
+    run.handle(
+        [
+            'secret-add',
+            '--label',
+            'mylabel',
+            '--description',
+            'mydesc',
+            '--expire',
+            '3d',
+            '--rotate',
+            'quarterly',
+            '--owner',
+            owner,
+            f'foo#file={mock_temp_dir}/foo',
+        ],
+        stdout='secretid',
+    )
+    result = hookcmds.secret_add(
+        {'foo': 'bar'},
+        label='mylabel',
+        description='mydesc',
+        expire='3d',
+        rotate=hookcmds.SecretRotate.QUARTERLY,
+        owner=owner,
+    )
+    assert result == 'secretid'
+
+
+def test_secret_add_date(run: Run, mock_temp_dir: str):
+    run.handle(
+        [
+            'secret-add',
+            '--expire',
+            '2025-12-31T23:59:59Z',
+            f'foo#file={mock_temp_dir}/foo',
+        ],
+        stdout='secretid',
+    )
+    result = hookcmds.secret_add(
+        {'foo': 'bar'},
+        expire=datetime.datetime(2025, 12, 31, 23, 59, 59, tzinfo=datetime.timezone.utc),
+    )
+    assert result == 'secretid'
+
+
+@pytest.mark.parametrize('peek,refresh', [(False, False), (False, True), (True, False)])
+def test_secret_get_id(run: Run, peek: bool, refresh: bool):
+    cmd = ['secret-get', '--format=json', 'secret:123']
+    if peek:
+        cmd.append('--peek')
+    if refresh:
+        cmd.append('--refresh')
+    run.handle(cmd, stdout='{"foo": "bar"}')
+    result = hookcmds.secret_get(id='secret:123', peek=peek, refresh=refresh)  # type: ignore
```

Comment:
Why is the `type: ignore` necessary here (and below)?

Reply:
The parametrisation ensures that the combinations for `peek` and `refresh` are both false or one true. pyright doesn't (from what I could tell) use that information, so all it has is that the signature for the test method is `bool` for both, so it 'thinks' that there's a possibility that both could be True. I'll add a comment.